### PR TITLE
Port Boards: Boards

### PR DESCRIPTION
**Pull request by dormi**
### Spec
We should be able, when clicking on Methodologies link on the sidebar, to see an index with our boards (or empty list if none). From there we should be to create a board, rename it, delete it.

**Proposed solution**
Port from Pro the desired functionality

### How to test

- Log into Dradis.
- Assert you can create a board.
- Assert you can rename it
- Assert you can delete it 

### Copyright assignment

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List

~~- [ ] Added a CHANGELOG entry~~

### CODE REVIEW

File: app/models/node.rb [hunk 1]
```diff
@@ -24,6 +24,10 @@ module Types
   has_many :issues, -> { distinct }, through: :evidence
   has_many :notes, dependent: :destroy
 
+  def boards
```
Comment (dormi) [inline app/models/node.rb]: we have `has_many` in line 22. We don't need this method?

File: app/models/project.rb [hunk 2]
```diff
@@ -47,6 +47,10 @@ def tags
     Tag.all
   end
 
+  def boards
```
Comment (dormi) [inline app/models/project.rb]: Alpha order these public instance methods

File: app/services/methodology_migration_job.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+class MethodologyMigrationJob < ApplicationJob
```
Comment (dormi) [inline app/services/methodology_migration_job.rb]: hey! this should be under `app/jobs` ?

File: app/services/methodology_migration_service.rb [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    @project = Project.find(project_id)
+  end
+
+  def migrate(methodology, board_name: nil, node: nil)
+    ActiveRecord::Base.transaction do
+      # create a board for each methodology
+      board = Board.new(
+        name: methodology.name,
+        node: node || project.methodology_library
+      )
```
Comment (dormi) [inline app/services/methodology_migration_service.rb]: I am not 100% sure we must remove the `project_id: project_id` line here.
If we replace it with `project: @project`, the line will be the same in both codebases.

File: app/services/methodology_migration_service.rb [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  # Returns 2 arrays, one of "pending" cards (not checked)
+  # and one of "done" cards (checked).
+  def extract_cards(methodology)
+    pending = []
+    done    = []
+    methodology.sections.each do |section|
+      section.tasks.each do |task|
+        # create a card for each task on the methodology
+        card = Card.new
+        card.name = "[#{section.name}] #{task.name}".truncate(Card.columns_hash['name'].limit || 256)
```
Comment (dormi) [inline app/services/methodology_migration_service.rb]: This `256` is new.
Is it a sqlite requirement yes? Because `Card.columns_hash['name'].limit` is `nil` in CE ?

File: config/routes.rb [hunk 6]
```diff
@@ -53,6 +53,15 @@
       end
     end
 
+    resources :boards do
```
Comment (dormi) [inline config/routes.rb]: place this in alpha order, it should go before `comments`
It would be in the same location as in Pro

File: spec/features/board_pages/board_pages_spec.rb [hunk 7]
```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
```
Comment (dormi) [inline spec/features/board_pages/board_pages_spec.rb]: This file seems quite different form Pro.
And maybe it should be more similar or ideally the same file.

Example: this code also works in CE: `create(:board, node: current_project.methodology_library, project: current_project)`

File: app/views/activities/_poll_methodologies.js.erb [hunk 8]
```diff
@@ -0,0 +1,51 @@
+<%# Card activities %>
```
Comment (dormi) [inline app/views/activities/_poll_methodologies.js.erb]: We are porting this, but never calling this partial. 
It should be from `activities/poll.js.erb`
But I think this belongs to another PR.

File: app/models/methodology.rb [hunk 9]
```diff
@@ -39,7 +43,7 @@ class Methodology
   # 'admin:paths:templates:methodologies' setting
   def self.pwd
     @pwd ||= begin
-      conf = Configuration.create_with(value: Rails.root.join('templates/methodologies/').to_s).
+      conf = Configuration.create_with(value: Rails.root.join('../../shared/templates/methodologies/').to_s).
```
Comment (etdsoft) [inline app/models/methodology.rb]: @brianp @dormi (!)

```suggestion
      conf = Configuration.create_with(value: Rails.root.join('templates/methodologies/').to_s).
```

File: spec/features/board_pages/board_pages_spec.rb (hunk 7 above)
Comment (etdsoft) [reply spec/features/board_pages/board_pages_spec.rb]: how is this being tracked so it doesn't slip if the PR is merged?

File: app/models/methodology.rb (hunk 9 above)
Comment (brianp) [reply app/models/methodology.rb]: It's weird that it works without this change 🤔

File: spec/features/board_pages/board_pages_spec.rb (hunk 7 above)
Comment (brianp) [reply spec/features/board_pages/board_pages_spec.rb]: > how is this being tracked so it doesn't slip if the PR is merged?

Notes/tasks to make sure in the next scope where they become appropriate they get re-added. It's an imperfect solution.

In this particular case the next scope already existed and we made sure the port existed in the next PR.

File: app/models/methodology.rb (hunk 9 above)
Comment (brianp) [reply app/models/methodology.rb]: I tested it and with the change everything still (continues?) to work as expected.
